src/gallium/drivers/lima/lima_resource.c

```diff
@@ -46,6 +46,35 @@ lima_resource_create(struct pipe_screen *pscreen,
 {
    struct lima_screen *screen = lima_screen(pscreen);
    struct lima_resource *res;
+   struct pipe_resource *pres;
+
+   if (screen->ro && templat->bind & PIPE_BIND_SCANOUT) {
+      struct pipe_resource scanout_templat = *templat;
+      struct renderonly_scanout *scanout;
+      struct winsys_handle handle;
+
+      scanout_templat.width0 = align(scanout_templat.width0, 16);
+      scanout_templat.height0 = align(scanout_templat.height0, 16);
+      scanout_templat.screen = pscreen;
+
+      scanout = renderonly_scanout_for_resource(&scanout_templat,
+                                                screen->ro, &handle);
+      if (!scanout)
+         return NULL;
+
+      assert(handle.type == DRM_API_HANDLE_TYPE_FD);
+      res = lima_resource(pscreen->resource_from_handle(pscreen, templat,
+                                                        &handle,
+                                                        PIPE_HANDLE_USAGE_WRITE));
+      close(handle.handle);
+      if (!res)
+         return NULL;
```

Comment:
Seems we need to free the scanout when import fail, and better to check the resource_from_handle() return pipe_resource * not lima_resource * in case we may change the offset of pipe_resource inside lima_resource.